sdk/src/main/scala/io/horizen/account/api/rpc/service/EthService.scala

```diff
@@ -1027,4 +1029,14 @@ class EthService(
       }
     }
   }
+
+  @RpcMethod("web3_sha3")
+  def getSHA3(data: Array[Byte]): Hash = {
+    try {
+      new Hash(Keccak256.hash(data))
+    } catch {
+      case e: IllegalArgumentException => throw new RpcException(RpcError.fromCode(RpcCode.InvalidParams, e.getCause.getMessage))
```

Comment:
Hm, I don't think there is any reasonable way this can fail. I don't think we should catch anything here, if it really does throw it will get turned into an internal server error - which would be correct here.

Reply:
Agree, error handling removed